tap_pendo/streams.py

```diff
@@ -20,15 +20,25 @@
 from singer.utils import now, strftime, strptime_to_utc
 from tap_pendo import utils as tap_pendo_utils
 
+
 KEY_PROPERTIES = ['id']
 US_BASE_URL = "https://app.pendo.io"
 EU_BASE_URL = "https://app.eu.pendo.io"
+API_RECORD_LIMIT = 100_000
 
 LOGGER = singer.get_logger()
 session = requests.Session()
 # timeout request after 300 seconds
 REQUEST_TIMEOUT = 300
 
+BACKOFF_FACTOR = 30
+
+def to_giveup(error):
+    """
+        Boolean function to return if we want to give up retrying based on error response
+    """
+    return error.response is not None and 400 <= error.response.status_code < 500
```

Comment:
Are there any chances where the response came without status_code? Because that isn't handled.